src/Sylius/Bundle/CoreBundle/Security/UserImpersonator.php

```diff
@@ -27,7 +27,7 @@ final class UserImpersonator implements UserImpersonatorInterface
     private string $firewallContextName;
 
     public function __construct(
-        private SessionInterface $session,
+        private RequestStack $requestStack,
```

Comment:
Shouldn't we make here a similar BC protection layer as we have [here](https://github.com/Sylius/Sylius/blob/symfony-6/src/Sylius/Bundle/CoreBundle/EventListener/TaxonDeletionListener.php#L49-L52)? 🤔

Reply:
Same for similar cases below